src/Operator/Constructors.jl

```diff
@@ -553,10 +692,181 @@ function diffusionoperator(basis::AbstractBasis, mesh::Mesh)
+    dy: 1.0
+
+2 inputs:
+operator field:
+  tensor product basis:
+    numbernodes1d: 3
+    numberquadraturepoints1d: 4
+    numbercomponents: 1
+    dimension: 2
+  evaluation mode:
+    interpolation
+operator field:
+  tensor product basis:
+    numbernodes1d: 3
+    numberquadraturepoints1d: 4
+    numbercomponents: 1
+    dimension: 2
+  evaluation mode:
+    quadratureweights
+   
+1 output:
+operator field:
+  tensor product basis:
+    numbernodes1d: 3
+    numberquadraturepoints1d: 4
+    numbercomponents: 1
+    dimension: 2
+  evaluation mode:
+    gradient
+```
+"""
+function advectionoperator(basis::AbstractBasis, mesh::Mesh)
+    # setup
+    U = 1.0
+    function advectionweakform(u::Array{Float64}, w::Array{Float64})
+        dv = U * u * w[1]
+        return [dv]
+    end
+
+    # fields
+    inputs = [
+        OperatorField(basis, [EvaluationMode.interpolation]),
+        OperatorField(basis, [EvaluationMode.quadratureweights]),
+    ]
+    outputs = [OperatorField(basis, [EvaluationMode.gradient])]
+
+    # operator
+    advection = Operator(advectionweakform, mesh, inputs, outputs)
+    return advection
+end
+
+"""
+```julia
+advectionmappedoperator(basis, mesh)
+```
+Convenience constructor for the mapped advection operator
+
+# Weak form:
+- ``\\int \\nabla v u``
+
+# Arguments:
+- `basis`: basis for all operator fields to use
+- `mesh`:  mesh for operator
+
+# Returns:
+- Mapped advection operator with basis on mesh
+
+# Example:
+```jldoctest
+# mapped advection operator
+mesh = Mesh2D(1.0, 1.0);
+basis = TensorH1LagrangeBasis(3, 4, 1, mesh.dimension)
+advectionmapped = LFAToolkit.advectionmappedoperator(basis, mesh);
+
+# verify
+println(advectionmapped)
+
+# output
+
+finite element operator:
+2d mesh:
+    dx: 1.0
+    dy: 1.0
+
+2 inputs:
+operator field:
+  tensor product basis:
+    numbernodes1d: 3
+    numberquadraturepoints1d: 4
+    numbercomponents: 1
+    dimension: 2
+  evaluation mode:
+    interpolation
+operator field:
+  tensor product basis:
+    numbernodes1d: 3
+    numberquadraturepoints1d: 4
+    numbercomponents: 1
+    dimension: 2
+  evaluation mode:
+    quadratureweights
+   
+1 output:
+operator field:
+  tensor product basis:
+    numbernodes1d: 3
+    numberquadraturepoints1d: 4
+    numbercomponents: 1
+    dimension: 2
+  evaluation mode:
+    gradient
+```
+"""
+function advectionmappedoperator(basis::AbstractBasis, mesh::Mesh)
+    # setup
+    h0 = 1.0
+    g = 9.81
+    U = sqrt(g * h0)
+    function advectionmappedweakform(u::Array{Float64}, w::Array{Float64})
+        dv = U * u * w[1]
+        return [dv]
+    end
+
+    # fields
+    inputs = [
+        OperatorField(basis, [EvaluationMode.interpolation]),
+        OperatorField(basis, [EvaluationMode.quadratureweights]),
+    ]
+    outputs = [OperatorField(basis, [EvaluationMode.gradient])]
+
+    # operator
+    advectionmapped = Operator(advectionmappedweakform, mesh, inputs, outputs)
+    return advectionmapped
+end
```

Comment:
@AdelekeBankole I think this can be deleted since it's no different from plain advection.

Reply:
ok, actually i think we should be careful how we handle both two examples: Because I created ex010 for advection and ex011 is for mapped advection. I will clean this up

Reply:
Right, the code here isn't an example and currently isn't used. But the mapped operator shouldn't be different. It's exactly the same weak form using a different basis.